### PR TITLE
[Ml-34704] Add old metrics like r2_score from using sklearn to directly use mlflow.metrics.make_metric

### PR DESCRIPTION
**Pull request by jessechancy**
## Related Issues/PRs
https://databricks.atlassian.net/browse/ML-34115

## What changes are proposed in this pull request?

Add regression and classification sklearn metrics to directly use mlflow.metrics.make_metric

## How is this patch tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

### CODE REVIEW

File: mlflow/metrics/__init__.py [hunk 1]
```diff
@@ -191,6 +197,53 @@
...
+
+# Regression Metrics
+#TODO: how to deal with sample weights input?
+#TODO: double check greater is better
+
+mae = make_metric(
+    eval_fn=_mae_eval_fn,
+    greater_is_better=False,
+    name="mae",
+    version="v1"
```
Comment (annzhang-db) [inline mlflow/metrics/__init__.py:209]: Sorry, forgot to mention this, we don't actually want to put a version on these metrics since we don't want to change the names of the metrics for backwards compatibility (with version the name will become mae/v1 instead of mae which it is now)

File: mlflow/metrics/utils/metric_definitions.py [hunk 2]
```diff
@@ -249,3 +249,64 @@ def _rougeLsum_eval_fn(eval_df, metrics):
...
+#TODO: add basic metrics definitions
+def _mae_eval_fn(eval_df, metrics):
+    if "target" in eval_df:
+        from sklearn.metrics import mean_absolute_error
+
+        y_pred = eval_df["prediction"]
+        y = eval_df["target"]
+
+        mae = mean_absolute_error(y, y_pred)
+        return MetricValue(aggregate_results={"": mae})
```
Comment (annzhang-db) [inline mlflow/metrics/utils/metric_definitions.py:262]: Also we decided to use the same name for aggregate name as metric instead of empty string: look [here](https://github.com/mlflow/mlflow/pull/9594/files#r1323668187)

**Review by annzhang-db — CHANGES_REQUESTED** [review]
Added two comments for now on some decisions we made that I forgot to mention - let me know if you have questions about either of them!
